### PR TITLE
Allow forcing backend with ExecHint

### PR DESCRIPTION
**Pull request by tomaka**
Fix https://github.com/paritytech/smoldot/issues/2341

The `ExecHint` now has two new variants that allow forcing usage of either wasmi or wasmtime.
This is useful for testing purposes and for fuzzing.

### CODE REVIEW

File: src/executor/vm/tests.rs [hunk 1]
```diff
@@ -25,42 +25,47 @@ fn is_send() {
...
+            .unwrap();
+
+        loop {
+            match vm.run(None) {
+                Ok(super::ExecOutcome::Finished {
+                    return_value: Ok(_),
+                }) => break,
+                Ok(super::ExecOutcome::Finished {
+                    return_value: Err(_),
+                }) => panic!(),
```
Comment (melekes) [inline src/executor/vm/tests.rs:57]: do you want to panic with error?

```suggestion
                }) => panic!("{:?}", e),
```

File: src/executor/vm/tests.rs [hunk 2]
```diff
@@ -25,42 +25,47 @@ fn is_send() {
...
+                Ok(super::ExecOutcome::Interrupted { id: 0, .. }) => break,
+                Ok(super::ExecOutcome::Interrupted { .. }) => panic!(),
+                Err(_) => panic!(),
```
Comment (melekes) [inline src/executor/vm/tests.rs:60]: same question

**Review by melekes — APPROVED** [review]
👍

File: src/executor/vm/tests.rs (hunk 1 above)
Comment (tomaka) [reply src/executor/vm/tests.rs:57]: Well, it's just a test :shrug:

Comment (melekes) [reply src/executor/vm/tests.rs:57]: Okay

**Pull request merged**